### PR TITLE
Implementation of lp:per-session-server-id bluepring

### PR DESCRIPTION
**Pull request by gl-sergei**
Added session variable pseudo_server_id with default value 0. When
pseudo_server_id is not zero, binlog events get logged with server_id
= pseudo_server_id.

Changing pseudo_server_id requires SUPER privilege.

Code change is trivial. Two MTR test added testing simple
master-slave setup and circular replication setup.

### CODE REVIEW

**Comment by gl-sergei** [comment]
http://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/919/

File: mysql-test/suite/sys_vars/r/all_vars.result [hunk 1]
```diff
@@ -16,5 +16,6 @@ INNODB_FAKE_CHANGES
 INNODB_FAKE_CHANGES
 INNODB_KILL_IDLE_TRANSACTION
 INNODB_KILL_IDLE_TRANSACTION
+PSEUDO_SERVER_ID
```
Comment (laurynas-biveinis) [inline mysql-test/suite/sys_vars/r/all_vars.result]: Why not a pseudo_server_id_basic.test ?

File: mysql-test/suite/rpl/t/pseudo_server_id.test [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+set session pseudo_server_id= 1;
+select @@session.pseudo_server_id;
+set session pseudo_server_id= default;
+select @@session.pseudo_server_id;
+
+CREATE USER 'user1'@'localhost' IDENTIFIED BY '111';
+connect(con_user1,localhost,user1,"111",,);
+--connection con_user1
+select @@session.pseudo_server_id;
+--error 1227
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rpl/t/pseudo_server_id.test]: Please use ER_FOO instead of a number

**Comment by laurynas-biveinis** [comment]
Looks good, just how about testing:
- setting pseudo_server_id and then server_id in the same thread;
- --replicate-same-server-id

I'd also considering something similar to mysql-test/include/show_binlog_events.inc, except that it wouldn't mask server_id, and adding it to the existing testcases.

**Comment by gl-sergei** [comment]
1. Added parameter `keep_server_id` for `show_binlog_events.inc` and added `show_binlog_events.inc` to existing tests.
2. Moved basic test and access rights test to `pseudo_server_id_basic` test
3. Added test setting `server_id` and `pseudo_server_id` in the same session
4. Added test with `replicate-same-server-id`

http://jenkins.percona.com/view/PS%205.6/job/percona-server-5.6-param/932/

**Pull request merged**